graylog2-server/src/main/java/org/graylog2/rest/resources/cluster/ClusterMetricsResource.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.rest.resources.cluster;
+
+import com.codahale.metrics.annotation.Timed;
+import com.wordnik.swagger.annotations.Api;
+import com.wordnik.swagger.annotations.ApiOperation;
+import com.wordnik.swagger.annotations.ApiParam;
+import com.wordnik.swagger.annotations.ApiResponse;
+import com.wordnik.swagger.annotations.ApiResponses;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.apache.shiro.authz.annotation.RequiresPermissions;
+import org.graylog2.cluster.Node;
+import org.graylog2.cluster.NodeNotFoundException;
+import org.graylog2.cluster.NodeService;
+import org.graylog2.rest.RemoteInterfaceProvider;
+import org.graylog2.rest.models.system.metrics.requests.MetricsReadRequest;
+import org.graylog2.rest.models.system.metrics.responses.MetricNamesResponse;
+import org.graylog2.rest.models.system.metrics.responses.MetricsSummaryResponse;
+import org.graylog2.shared.rest.resources.RestResource;
+import org.graylog2.shared.rest.resources.system.RemoteMetricsResource;
+import org.graylog2.shared.security.RestPermissions;
+import retrofit.Response;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.util.List;
+
+@RequiresAuthentication
+@Api(value = "Cluster/Metrics", description = "Cluster-wide Internal Graylog2 metrics")
+@Path("/cluster/{nodeId}/metrics")
+public class ClusterMetricsResource extends RestResource {
+    private final RemoteMetricsResource remoteMetricsResource;
+
+    @Inject
+    public ClusterMetricsResource(@PathParam("nodeId") String nodeId,
```

Comment:
There is no way of setting the `nodeId` from the swagger UI. It looks like swagger only generates an input when the parameter is in the resource actions.
The parameter is also missing the `@ApiParam` annotation.